opencti-platform/opencti-graphql/src/database/stix-2-0-converter.ts

```diff
@@ -0,0 +1,136 @@
+import { ENTITY_TYPE_CONTAINER_FEEDBACK } from '../modules/case/feedback/feedback-types';
+import { ENTITY_TYPE_CONTAINER_TASK } from '../modules/task/task-types';
+import { ENTITY_TYPE_CONTAINER_CASE_INCIDENT } from '../modules/case/case-incident/case-incident-types';
+import { ENTITY_TYPE_CONTAINER_CASE_RFI } from '../modules/case/case-rfi/case-rfi-types';
+import { ENTITY_TYPE_CONTAINER_CASE_RFT } from '../modules/case/case-rft/case-rft-types';
+import { getFileContent } from './file-storage';
+
+export const convertTypeToStix2Type = (type: string): string => {
+  if (isStixDomainObjectIdentity(type)) {
+    return 'identity';
+  }
+  if (isStixDomainObjectLocation(type)) {
+    return 'location';
+  }
+  if (type.toLowerCase() === ENTITY_HASHED_OBSERVABLE_STIX_FILE.toLowerCase()) {
+    return 'file';
+  }
+  if (isStixCoreRelationship(type)) {
+    return 'relationship';
+  }
+  if (isStixSightingRelationship(type)) {
+    return 'sighting';
+  }
+  if (isStixDomainObjectThreatActor(type)) {
+    return 'threat-actor';
+  }
+  if (type === ENTITY_TYPE_CONTAINER_CASE_INCIDENT
+    || type === ENTITY_TYPE_CONTAINER_CASE_RFI
+    || type === ENTITY_TYPE_CONTAINER_CASE_RFT
+    || type === ENTITY_TYPE_CONTAINER_FEEDBACK
+    || type === ENTITY_TYPE_CONTAINER_TASK) {
+    return `x-opencti-${type.toLowerCase()}`;
+  }
+  if (type === ENTITY_TYPE_DATA_COMPONENT || type === ENTITY_TYPE_DATA_SOURCE) {
+    return `x-mitre-${type.toLowerCase()}`;
+  }
+  return type.toLowerCase();
+};
+
+// Meta
+const buildKillChainPhases = (instance: StoreEntity | StoreRelation): Array<SMO.StixInternalKillChainPhase> => {
+  return (instance[INPUT_KILLCHAIN] ?? []).map((k) => {
+    const data: SMO.StixInternalKillChainPhase = {
+      kill_chain_name: k.kill_chain_name,
+      phase_name: k.phase_name,
+      x_opencti_order: k.x_opencti_order,
+    };
+    return cleanObject(data);
+  });
+};
+
+const buildExternalReferences = (instance: StoreObject): Array<SMO.StixInternalExternalReference> => {
+  return (instance[INPUT_EXTERNAL_REFS] ?? []).map((e) => {
+    const data: SMO.StixInternalExternalReference = {
+      source_name: e.source_name,
+      description: e.description,
+      url: e.url,
+      hash: e.hashes,
+      external_id: e.external_id,
+    };
+    return cleanObject(data);
+  });
+};
+
+// Builders
+const buildStixObject = async (instance: StoreObject): Promise<S.StixObject> => {
+  return {
+    id: instance.standard_id,
+    x_opencti_id: instance.id,
+    spec_version: '2.0',
+    x_opencti_type: instance.entity_type,
+    type: convertTypeToStix2Type(instance.entity_type),
+    x_opencti_granted_refs: (instance[INPUT_GRANTED_REFS] ?? []).map((m) => m.standard_id),
+    x_opencti_workflow_id: instance.x_opencti_workflow_id,
+    x_opencti_files: await Promise.all((instance.x_opencti_files ?? []).map(async (file: StoreFileWithRefs) => ({
+      name: file.name,
+      data: await getFileContent(file.id, 'base64'),
```

Comment:
Are you sure about this? I dont think the file data content should be exposed like this

Reply:
In pycti, x_opencti_files always have the data attached. It may be better to remove it, but we won't keep the same behavior as pycti if we do so. Do we prioritize keeping the same behavior to prevent breaking changes, or do we remove it to improve export format?